### PR TITLE
[5.4] Allow files to be sent on post, put and patch methods

### PR DESCRIPTION
**Pull request by mateusjatenee**
Hi folks,  

One thing I dislike when testing is having to use the `call` method whenever you want to send files. This PR allows files to be sent on the `post`, `put` and `patch` methods. I added it as the last argument so it does not break any tests.  

My editor messed up with the styling but I'm sure StyleCI will fix it should it be approved.

Let me know what you guys think!  

Cheers

### CODE REVIEW

File: src/Illuminate/Foundation/Testing/Concerns/MakesHttpRequests.php [hunk 1]
```diff
@@ -2,12 +2,12 @@
 
 namespace Illuminate\Foundation\Testing\Concerns;
 
-use Illuminate\Support\Str;
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Testing/Concerns/MakesHttpRequests.php]: please revert all these changes

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Illuminate/Foundation/Testing/Concerns/MakesHttpRequests.php [hunk 2]
```diff
@@ -253,8 +253,8 @@ protected function prepareUrlForRequest($uri)
             $uri = substr($uri, 1);
         }
 
-        if (! Str::startsWith($uri, 'http')) {
-            $uri = config('app.url').'/'.$uri;
+        if (!Str::startsWith($uri, 'http')) {
+            $uri = config('app.url') . '/' . $uri;
```
Comment (taylorotwell) [inline src/Illuminate/Foundation/Testing/Concerns/MakesHttpRequests.php]: Please don't make these unrelated changes.

File: src/Illuminate/Foundation/Testing/Concerns/MakesHttpRequests.php [hunk 3]
```diff
@@ -274,8 +274,8 @@ protected function transformHeadersToServerVars(array $headers)
         foreach ($headers as $name => $value) {
             $name = strtr(strtoupper($name), '-', '_');
 
-            if (! Str::startsWith($name, $prefix) && $name != 'CONTENT_TYPE') {
-                $name = $prefix.$name;
+            if (!Str::startsWith($name, $prefix) && $name != 'CONTENT_TYPE') {
+                $name = $prefix . $name;
```
Comment (taylorotwell) [inline src/Illuminate/Foundation/Testing/Concerns/MakesHttpRequests.php]: We don't use this format anywhere in the framework, also an unrelated change.

**Comment by taylorotwell** [comment]
You can re-submit this without all the other unneeded changes you made if you want.

File: src/Illuminate/Foundation/Testing/Concerns/MakesHttpRequests.php (hunk 3 above)
Comment (mateusjatenee) [inline src/Illuminate/Foundation/Testing/Concerns/MakesHttpRequests.php]: yeah, i know — it's from the plugin I use. I thought StyleCI could fix this. I'll resubmit this later.  

Thanks
